cql2pgjson/src/main/java/org/folio/cql2pgjson/CQL2PgJSON.java

```diff
@@ -756,6 +765,30 @@ private String pgId(CQLTermNode node) throws QueryValidationException {
     }
   }
 
+  private String arrayNode(String index, CQLTermNode node, org.folio.rest.persist.ddlgen.Modifier relationModifier,
+    String modifierValue, DbIndex dbIndex, CqlModifiers modifiers) throws QueryValidationException {
+
+    StringBuilder res = new StringBuilder();
+
+    IndexTextAndJsonValues vals = new IndexTextAndJsonValues();
+    vals.setIndexText(index2sqlText("t.c", relationModifier.getSubfield()));
+
+    final String table = this.jsonField.split("\\.")[0];
+    final String field = this.jsonField.split("\\.")[1];
+
+    res.append("id in (select t.id from (select id as id, jsonb_array_elements(");
+    res.append(field + "->'" + index + "') as c from " + table + ") as t where t.c");
+    res.append(" @> '{\"");
```

Comment:
This SQL is hard to read and understand.

How about something like this with indentation that follows the structure?

    String modifierName = Cql2SqlUtil.cql2string(relationModifier.getModifierName());
    ...
    String res =
        "id in (select t.id "
      + "  from (select id as id,"
      + "               jsonb_array_elements(" + index + ") as c"
      + "        from " + table ") as t"
      + "  where t.c @> '{\"" + modifierName + "\": \"" + modifierValue + "\"}'"
      + "        and " + indexNode
      + ")"
    );

Reply:
ok. Something changed.